test/model/compose_test.dart

```diff
@@ -0,0 +1,190 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/model/compose.dart';
+
+void main() {
+  group('wrapWithBacktickFence', () {
+    /// Check `wrapWithBacktickFence` on example input and expected output.
+    ///
+    /// The intended input (content passed to `wrapWithBacktickFence`)
+    /// is straightforward to infer from `expected`.
+    /// To do that, this helper takes `expected`, validates it
+    /// (as a courtesy for the test author), and removes the opening and
+    /// closing fences.
+    ///
+    /// Then we have the input to the test, as well as the expected output.
+    void checkFenceWrap(String expected, {String? infoString}) {
+      final lines = expected.split('\n');
+      final firstLineMatch = RegExp(r'^(`{3,})[^`]*$').allMatches(lines.removeAt(0)).singleOrNull;
+      assert(firstLineMatch != null,
+        'test error: opening fence not found in `expected`');
+      assert(lines.removeAt(lines.length - 1) == '',
+        'test error: `expected` should end with a newline');
+      final openingFenceLength = firstLineMatch![1]!.length;
+      assert(RegExp(r'^`{' + openingFenceLength.toString() + r'}$').hasMatch(lines.removeAt(lines.length - 1)),
+        'test error: closing fence not found in `expected`');
+      final content = lines.join('\n');
+      check(wrapWithBacktickFence(content: content, infoString: infoString)).equals(expected);
+    }
+
+    test('single line with no code blocks', () {
```

Comment:
One other set of scenarios that would be good to test: empty content; content consisting of blank lines.